src/internal_modules/roc_core/heap_arena.h

```diff
@@ -30,7 +39,11 @@ namespace core {
 //! The memory is always maximum aligned. Thread-safe.
 class HeapArena : public IArena, public NonCopyable<> {
 public:
-    HeapArena();
+    //! Initialize.
+    //!
+    //! @b Parameters
+    //! - @p flags defines options to modify behaviour as indicated in HeapArenaFlags
+    HeapArena(size_t flags = DefaultHeapArenaFlags);
```

Comment:
In HeapArena, we have static method enable_leak_detection() which enabled panic on leak for all arena instances. (This method is used to enable panics globally in tools and tests, but to disable it in C library.)

With this constructor, we now have two source of configuration for arena: one via static method, another via ctor flags.

To keep things simpler, I'd suggest to keep only one:

* remove flags from ctor
* replace static method enable_leak_detection() with static method set_flags() or configure() or something like this, which accepts flags
* define two flags: EnableGuards and EnableLeakDetection
* initially flags are set to EnableGuards
* in places where we called enable_leak_detection(), we will now set flags EnableGuards | EnableLeakDetection
* in tests we can temporary change flags and then change them back

Reply:
Along with the above... I've kept the DefaultHeapArenaFlags, and used that all over as well. Let me know.

Reply:
LGTM